service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/ainotebook/ControlledAiNotebookHandler.java

```diff
@@ -71,7 +71,7 @@ public String generateCloudName(@Nullable UUID workspaceUuid, String aiNotebookN
         generatedName.length() > MAX_INSTANCE_NAME_LENGTH
             ? generatedName.substring(0, MAX_INSTANCE_NAME_LENGTH)
             : generatedName;
-    generatedName = generatedName.replaceAll("[^a-zA-Z0-9-]+|^[^a-zA-Z0-9]+|[^a-zA-Z0-9]+$", "");
+    generatedName = generatedName.replaceAll("[^a-z0-9-]+|^[^a-z]+|[^a-z0-9]+$", "");
```

Comment:
what happened if it leads with a dash?

Can you also add a comment here explaining the regex?

Reply:
it will trim the start with dash. Yes. add comments to all resource reg